File: algorithm/src/main/java/baekjoon/p12760/Main.java
package baekjoon.p12760;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] inputs = new int[n][m];

        for (int i = 0; i < inputs.length; i++) {
            for (int j = 0; j < inputs[i].length; j++) {
                inputs[i][j] = sc.nextInt();
            }
        }

        for (int solve : solve(inputs, n, m)) {
            System.out.print(solve + " ");
        }
    }

    private static List<Integer> solve(int[][] inputs, int n, int m) {
        for (int[] e : inputs) {
            Arrays.sort(e);
        }

        int[] score = new int[n];
        for (int i = m - 1; i >= 0; i--) {
            int[] temp = new int[n];
            for (int j = 0; j < temp.length; j++) {
                temp[j] = inputs[j][i];
            }

            int max = max(temp);
            for (int k = 0; k < score.length; k++) {
                if (inputs[k][i] == max) {
                    score[k]++;
                }
            }
        }

        List<Integer> ret = new ArrayList<>();
        int max = max(score);
        for (int i = 0; i < score.length; i++) {
            if (score[i] == max) {
                ret.add(i + 1);
            }
        }

        return ret;
    }

    private static int max(int[] arr) {
        int ret = arr[0];
        for (int e : arr) {
            ret = Math.max(ret, e);
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p12760/Main.kt
package baekjoon.p12760

fun main(args: Array<String>) {
    val (n, m) = readLine()!!.trim().split(" ").map { it.toInt() }
    val inputs = List(n) { readLine()!!.trim().split(" ").map { it.toInt() } }
    println(solve(inputs, n, m).joinToString(" "))
}

private fun solve(inputs: List<List<Int>>, n: Int, m: Int): List<Int> {
    val descending = (0 until n).map { inputs[it].sortedDescending() }
    val score = IntArray(n)

    for (i in 0 until m) {
        val temp = (0 until n).map { descending[it][i] }
        val max = temp.max()

        for (j in 0 until n) {
            if (descending[j][i] == max) {
                score[j]++
            }
        }
    }

    val max = score.max()
    return (0 until n).filter { score[it] == max }.map { it.inc() }
}